Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/components/SoundComponent.java

```diff
@@ -0,0 +1,109 @@
+package pg.autyzm.friendly_plans.view.components;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.view.View;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+import database.entities.Asset;
+import database.repository.AssetRepository;
+import java.io.IOException;
+import javax.inject.Inject;
+import pg.autyzm.friendly_plans.AppComponent;
+import pg.autyzm.friendly_plans.R;
+import pg.autyzm.friendly_plans.asset.AssetsHelper;
+import pg.autyzm.friendly_plans.notifications.ToastUserNotifier;
+
+public final class SoundComponent {
+
+    @Inject
+    MediaPlayer mediaPlayer;
+    @Inject
+    AssetRepository assetRepository;
+
+    private AssetsHelper assetsHelper;
+    private Long soundId;
+    private ImageView playSoundIcon;
+    private Context context;
+
+    public static SoundComponent getSoundComponent(Long soundId, ImageView playSoundIcon,
+            Context context, AppComponent appComponent) {
+        final SoundComponent soundComponent = new SoundComponent(soundId, playSoundIcon, context);
```

Comment:
Are you sure it does what your intention was?

Reply:
The final is needed because of referencing soundComponent from an anonymous class which implements MediaPlayer.OnCompletionListener
I know it's a clumsy construction but I couldn't find a better solution...

Reply:
Ok, fine for me.